mantidproject/mantid
# fix cppcheck warnings

**Description of work.**
Fixes a big chunk of cppcheck defects and dumps the rest into the supression file so that the threshold can be set to zero so that we have fewer merge conflicts.

**To test:**
- Review the code changes. Although a large proportion of the changes are trivial name changes (e.g. shadow parameter warnings), there are a few small tweaks to logic that may change behaviour.
- Verify that cppcheck job ran and wasn't skipped.

*This does not require release notes* because there should not be any detectable changes for the end user.

Fixes #32294 



---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
